File: db/csharp/SalesManagement.Domain/Models/PurchaseOrderDetail.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 発注データ明細
    /// </summary>
    public class PurchaseOrderDetail
    {
        public string PoNo { get; set; } = string.Empty;
        public int PoRowNo { get; set; }
        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public int UnitPrice { get; set; }
        public int Quantity { get; set; }
        public int ConsumptionTaxRate { get; set; }
        public int ReceivedQuantity { get; set; }
        public int Discount { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: db/fsharp/SalesManagement.Domain/Models/PurchaseOrderDetail.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 発注データ明細
/// </summary>
[<CLIMutable>]
type PurchaseOrderDetail = {
    PoNo: string                   // 発注番号（PK、FK）
    PoLineNo: int                  // 発注行番号（PK）
    ProductCode: string            // 商品コード（FK）
    ProductName: string            // 商品名（履歴保持）
    PoQuantity: int                // 発注数量
    ReceivedQuantity: int          // 入荷済数量
    UnitPrice: int                 // 仕入単価（履歴保持）
    Amount: int                    // 金額
    ConsumptionTax: int            // 消費税
    CreatedAt: DateTime            // 作成日時
    CreatedBy: string              // 作成者名
    UpdatedAt: DateTime            // 更新日時
    UpdatedBy: string              // 更新者名
}
